spec/lib/jobs/dell-racadm-set-bios-job-spec.js

```diff
@@ -0,0 +1,194 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+describe(require('path').basename(__filename), function () {
+    var uuid;
+    var RacadmSetBiosJob;
+    var Promise;
+    var job;
+    var encryption;
+    var lookup;
+    var racadmTool;
+    var mockWaterline = {
+        nodes: {},
+        catalogs: {}
+    };
+
+    before(function() {
+        helper.setupInjector([
+            helper.require('/lib/jobs/base-job'),
+            helper.require('/lib/jobs/dell-racadm-set-bios-job'),
+            helper.require('/lib/utils/job-utils/racadm-tool.js'),
+            helper.require('/lib/utils/job-utils/racadm-parser.js'),
+            helper.di.simpleWrapper(mockWaterline, 'Services.Waterline')
+        ]);
+        Promise = helper.injector.get('Promise');
+        RacadmSetBiosJob = helper.injector.get('Job.DellRacadm.SetBIOS');
+        racadmTool = helper.injector.get('JobUtils.RacadmTool');
+        uuid = helper.injector.get('uuid');
+        lookup = helper.injector.get('Services.Lookup');
+        encryption = helper.injector.get('Services.Encryption');
+    });
+
+    describe('Input validation', function() {
+        beforeEach('Dell Racadm Tool Set BIOS Validation', function() {
+            var options = {
+                username:"onrack",
+                password: "onrack",
+                filePath:"//1.2.3.4/src/bios.xml"
+            };
+            job = new RacadmSetBiosJob(options, {}, uuid.v4());
+            mockWaterline.nodes.findByIdentifier = function(){};
+            this.sandbox = sinon.sandbox.create();
+            this.sandbox.stub(job, '_subscribeActiveTaskExists').resolves();
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier');
+        });
+
+        afterEach('Dell Racadm Tool Set BIOS Validation', function() {
+            this.sandbox.restore();
+        });
+
+        it('should fail if node does not exist', function(done) {
+            mockWaterline.nodes.findByIdentifier.resolves(null);
+            job.run()
+                .then(function() {
+                    done(new Error("Expected job to fail"));
+                })
+                .catch(function(e) {
+                    try {
+                        expect(e).to.have.property('name').that.equals('AssertionError');
+                        expect(e).to.have.property('message').that.equals('No node for dell' +
```

Comment:
Since `e` is an instance of Error, so it's not necessary to check its property `name` and `message`, so to check its Error type and message is enough.
Meanwhile, to expect a promise to rejected, it can be as simple as following:
```
return expect(job.run()).to.be.rejectedWith(AssertionError, 'No node for dell racadm set bios');
```